### PR TITLE
fish: 3.3.1 -> 3.4.0

### PR DESCRIPTION
**Pull request by kidonng**
###### Description of changes

https://github.com/fish-shell/fish-shell/releases/tag/3.4.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
Hm, I'm seeing tests fail on aarch64-linux. ofborg shows:

```
The following tests FAILED:
        209 - exit_handlers.py (Failed)
```

and building it on my x86 machine using qemu-aarch64 shows:

```
The following tests FAILED:                                                            
        153 - noshebang.fish (Failed)                                                  
        171 - sigint.fish (Failed)                                                     
```

Do you know if there are any issues for this upstream?

**Review by thiagokokada — APPROVED** [review]
Result of `nixpkgs-review pr 163886` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>fish</li>
    <li>kitty</li>
    <li>oh-my-fish</li>
  </ul>
</details>

Tested and seems to work fine.

**Comment by cole-h** [comment]
Once this makes it into unstable, anybody who uses fish on aarch64-linux will be unable to update their system.

I'm going to revert (https://github.com/NixOS/nixpkgs/pull/163904) and we should see if upstream has any ideas why these tests might be failing (and why there are different tests failing, too).

I'd also appreciate if you would wait a little longer for maintainer approvals, if possible. As the sole maintainer of this package, I was looking into the problem, but also left a comment stating that something was broken.

**Comment by thiagokokada** [comment]
> I'd also appreciate if you would wait a little longer for maintainer approvals, if possible. As the sole maintainer of this package, I was looking into the problem, but also left a comment stating that something was broken.

Sorry, didn't saw your comment before merging because I had this tab opened for quite sometime and your comment didn't show to me. If you use GitHub for quite sometime you should know how the UI of GitHub gets broken sometimes (for example, I was seeing only my comment when I merged this PR).

**Comment by cole-h** [comment]
@kidonng Would you mind resubmitting this PR? Please and thank you.

**Pull request merged**
